posthog/hogql/ast.py

```diff
@@ -303,15 +312,23 @@ def has_child(self, name: str, context: HogQLContext) -> bool:
 
 
 @dataclass(kw_only=True)
-class IntegerType(ConstantType):
+class NumericType(ConstantType):
+    data_type: ConstantDataType = field(default="numeric", init=False)
+
+    def print_type(self) -> str:
+        return "Numeric"
```

Comment:
Should we do a `raise "should always be float or int"` here to prevent awkward bugs (and introduce few new ones)?

Context: I hate JavaScript's "numeric" type with a passion, and I don't want to introduce one to HogQL if we can at all avoid it. It just makes everything more complex. Keeping it to just "float" and "int" is more than enough. If this is an abstract class that's only used for signatures, then I have nothing against it (let's just make it unusable directly)... but even then, it might really matter for some functions if it's a Int or a Float, and perhaps we should just be really explicit with them in the signatures, e.g. when computing the return type?